# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBaselineCompile.cpp
# Commit: c414a69233f0
# Full Hash: c414a69233f0339ee83cde1f2407d263586038da
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-30 03:48:30
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 7: Remove Frame::tls. r=lth
#   
#   We remove Frame::tls and fix prologue/epilogue. Runtime, Ion and Baseline are ready for this move already.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83064
# ==============================================================================

diff -r 1e72f84b4206 -r c414a69233f0 js/src/wasm/WasmBaselineCompile.cpp
--- a/js/src/wasm/WasmBaselineCompile.cpp	Fri Oct 23 18:17:38 2020 +0000
+++ b/js/src/wasm/WasmBaselineCompile.cpp	Fri Oct 23 18:17:38 2020 +0000
@@ -5235,9 +5235,7 @@
   // Labels
 
   void insertBreakablePoint(CallSiteDesc::Kind kind) {
-    // The debug trap exit requires WasmTlsReg be loaded. However, since we
-    // are emitting millions of these breakable points inline, we push this
-    // loading of TLS into the FarJumpIsland created by linkCallSites.
+    fr.loadTlsPtr(WasmTlsReg);
     masm.nopPatchableToCall(CallSiteDesc(iter_.lastOpcodeOffset(), kind));
   }
 
@@ -5576,6 +5574,9 @@
       restoreRegisterReturnValues(resultType);
     }
 
+    // To satisy Tls extent invariant we need to reload WasmTlsReg because
+    // baseline can clobber it.
+    fr.loadTlsPtr(WasmTlsReg);
     GenerateFunctionEpilogue(masm, fr.fixedAllocSize(), &offsets_);
 
 #if defined(JS_ION_PERF)